### PR TITLE
Add feature: Provider can Stake 

### PR DESCRIPTION
**Pull request by mattheworris**
Implementation of feature to allow staking tokens to selected Provider

- [x] Add tests for staking

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,8 +1,13 @@
...
 
-integration-test:
-	./scripts/run_integration_tests.sh
+tests-all: tests-unit tests-e2e
 
-integration-test-only:
-	./scripts/run_integration_tests.sh -s
+tests-unit:
+	@echo "---------------------------------------------"
+	@echo "Running unit tests"
```
Comment (shannonwells) [inline Makefile:7]: 👍

File: src/components/Stake.svelte [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+      selectLabel="Key to Stake From"
+      bind:selectedOption={selectedKey}
+      {validAccounts}
+    />
+    <label for="stakingInput">Amount to Stake in Tokens</label>
+    <div class="input-container">
+      <input type="number" id="stakingInput" value="1" style="text-align: right;" on:input={handleInput} />
+      <span class="units">{token}</span>
+    </div>
+    <button on:click={stake}>Stake</button>
```
Comment (shannonwells) [inline src/components/Stake.svelte:133]:
```suggestion
    <button on:click|preventDefault={stake}>Stake</button>
```
This is the (Svelte) Way

File: src/components/Stake.svelte [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+  const stake = async (evt: Event) => {
+    evt.preventDefault();
+    clearTxnStatuses();
+    let endpointURI: string = thisDotApi.selectedEndpoint || '';
+    if (selectedKey === '') {
+      alert('Please choose a key to stake from.');
+    } else if (isFunction(thisWeb3FromSource) && isFunction(thisWeb3Enable)) {
+      let signingKeys = validAccounts[selectedKey];
+      showTransactionStatus = true;
+      if (isLocalhost(endpointURI)) {
```
Comment (shannonwells) [inline src/components/Stake.svelte:72]: Non blocking:  I realized we're actually doing the web3Enable and web3FromSource checks before checking for localhost, when we don't need to.  In my PR I move these checks inside the `else` below for when we use an extension.

**Review by shannonwells — APPROVED** [review]
FYI, Cancel Stake doesn't hide the form, but since we're changing how all that's done in my PR, I don't think you should bother fixing it.

File: src/components/Stake.svelte (hunk 2 above)
Comment (mattheworris) [reply src/components/Stake.svelte:133]: Ah, right. I knew that, but chatGPT led me astray.

File: src/components/Stake.svelte (hunk 3 above)
Comment (mattheworris) [reply src/components/Stake.svelte:72]: Moved.

**Comment by mattheworris** [comment]
> FYI, Cancel Stake doesn't hide the form, but since we're changing how all that's done in my PR, I don't think you should bother fixing it.

That's strange it's working for me. I will note to add a test for that too.

**Pull request merged**